src/Bridges/slackbridge.jl

```diff
@@ -0,0 +1,162 @@
+# scalar version
+
+"""
+    ScalarSlackBridge{T, F, S}
+
+The `ScalarSlackBridge` converts a constraint `F`-in-`S` where `F` is a function different
+ from `SingleVariable` into the constraints ``F in EqualTo{T}`` and `SingleVariable`-in-`S`.
+"""
+struct ScalarSlackBridge{T, F<:MOI.AbstractScalarFunction, S<:MOI.AbstractScalarSet} <: AbstractBridge
+    slack::MOI.VariableIndex
+    slack_in_set::CI{MOI.SingleVariable, S}
+    equality::CI{F, MOI.EqualTo{T}}
+end
+function ScalarSlackBridge{T, F, S}(model, f::F, s::S) where {T, F<:MOI.AbstractScalarFunction, S<:MOI.AbstractScalarSet}
+    slack = MOI.add_variable(model)
+    new_f = MOIU.operate(-, T, f, MOI.SingleVariable(slack))
+    slack_in_set = MOI.add_constraint(model, MOI.SingleVariable(slack), s)
+    equality = MOI.add_constraint(model, new_f, MOI.EqualTo(0.0))
+    return ScalarSlackBridge{T, F, S}(slack, slack_in_set, equality)
+end
+
+# start allowing everything (scalar)
+MOI.supports_constraint(::Type{ScalarSlackBridge{T, F, S}},
+                        ::Type{<:MOI.AbstractScalarFunction},
+                         ::Type{<:MOI.AbstractScalarSet}) where {T, F, S} = true
+# then restrict (careful with ambiguity)
+MOI.supports_constraint(::Type{ScalarSlackBridge{T, F, S}},
+                        ::Type{<:MOI.SingleVariable},
+                        ::Type{<:MOI.EqualTo}) where {T, F, S} = false
+MOI.supports_constraint(::Type{ScalarSlackBridge{T, F, S}},
+                        ::Type{<:MOI.SingleVariable},
+                        ::Type{<:MOI.AbstractScalarSet}) where {T, F, S} = false
+MOI.supports_constraint(::Type{ScalarSlackBridge{T, F, S}},
+                        ::Type{<:MOI.AbstractScalarFunction},
+                        ::Type{<:MOI.EqualTo}) where {T, F, S} = false
+function added_constraint_types(::Type{ScalarSlackBridge{T, F, S}}) where {T, F, S}
+    return [(F, MOI.EqualTo{T}), (MOI.SingleVariable, S)]
```

Comment:
F should be `promote_operation(-, T F, SingleVariable)`

Reply:
Using just `F` is sufficient for the three scalar functions types that we have in MOI but this bridge might be used by functions defined outside of MOI for which a `promote_operation` is necessary

Reply:
By the way, this function is not being hit, should I test it explicitly?

Reply:
Actually here it should still be F with no promote_operation, such promotion occurred before creating the bridge instance.

Reply:
Yes but it needs to determine the constraints being created

Reply:
Both `concrete_bridge_type` and `added_constraint_types` are called before the bridge constructor?
I agree that we need `promote_operation` inside `concrete_bridge_type`.
But I don't think it should be inside `added_constraint_types`, just like the other bridges.

Reply:
You are right `F` is already the type computed with `promote_operation` in `conrete_type_bridge` so no need to change it